### PR TITLE
update-custom-test-image-template-nested

### PR DESCRIPTION
**Pull request by sallyom**
might be easier to merge, or cherry-pick the commit
the 'enable-nested-virt' script needed updates

### CODE REVIEW

File: ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml [hunk 1]
```diff
@@ -39,6 +39,5 @@ tests:
     ./hack/ci/test.sh
   openshift_installer_custom_test_image:
     cluster_profile: gcp
-    from: pipeline:kubevirt-test
+    from: stable:kubevirt-test
```
Comment (dhiller) [inline ci-operator/config/kubevirt/kubevirt/kubevirt-kubevirt-master.yaml:42]: Can you explain why this needs to be done, please? I always thought that during the run the test image would be built as part of the pipeline and therefore it should be referenced with pipeline? Don't we get another image otherwise?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml [hunk 2]
```diff
@@ -35,7 +34,8 @@ parameters:
 - name: CLUSTER_VARIANT
 - name: BUILD_ID
   required: false
-- name: USE_LEASE_CLIENT
+# TODO: uncomment - comment only for testing
+#- name: USE_LEASE_CLIENT
```
Comment (dhiller) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml:37]: Doesn't this need to be uncommented as per the comment above?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml [hunk 3]
```diff
@@ -342,7 +342,7 @@ objects:
       terminationMessagePolicy: FallbackToLogsOnError
       volumeMounts:
       - name: shared-tmp
-        mountPath: /tmp/shared
+        mountPath: /tmp
```
Comment (dhiller) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml]: I would really prefer to change it to `/tmp/shared` everywhere in the template for consistency?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml [hunk 4]
```diff
@@ -20,7 +20,6 @@ parameters:
 - name: CLUSTER_TYPE
   required: true
 - name: CLUSTER_ENABLE_NESTED_VIRT
-  value: "false"
```
Comment (dhiller) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml]: Why is this required?

File: ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml [hunk 5]
```diff
@@ -391,19 +391,21 @@ objects:
...
         set -euo pipefail
 
-        trap 'rc=$?; if test "${rc}" -eq 0; then touch /tmp/shared/setup-success; else touch /tmp/shared/exit /tmp/shared/setup-failed; fi; exit "${rc}"' EXIT
+        trap 'rc=$?; if test "${rc}" -eq 0; then touch /tmp/setup-success; else touch /tmp/exit /tmp/setup-failed; fi; exit "${rc}"' EXIT
         trap 'CHILDREN=$(jobs -p); if test -n "${CHILDREN}"; then kill ${CHILDREN} && wait; fi' TERM
 
-        if [[ "${CLUSTER_TYPE}" != "gcp" ]] && [[ "${CLUSTER_ENABLE_NESTED_VIRT-false}" == "true" ]]; then
-            echo "CLUSTER_ENABLE_NESTED_VIRT is unsupported for cluster type '${CLUSTER_TYPE}'"
-            exit 1
+        if [[ "${CLUSTER_ENABLE_NESTED_VIRT}" ]]; then
```
Comment (dhiller) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-custom-test-image.yaml:398]: Why did you have to change it back to not using the `${-}` syntax? I guess this will not work with using the `set -u`? I'd prefer not to use nested ifs to reduce complexity for the reader.

**Comment by sallyom** [comment]
closing, working w/ dhiller to incorporate these changed :)
